### ISSUE TITLE
Should getUserMedia promise be resolved only on pages that have focus

### ISSUE THREAD

**Issue by youennf** [w3c/mediacapture-main#560]
Safari currently does not resolve/reject a getUserMedia promise until the view is visible.
This allows to start capture on a page that the user has access to.
Should this be mandated in the spec?
The same question might apply to enumerateDevices.

**Comment by jan-ivar** [comment]
Firefox has this behavior as well. Turning on camera/mic from a background tab is creepy.

This might even be something that could go in the [permissions](https://w3c.github.io/permissions/#request-permission-to-use) spec.

Having prompts appear from background tabs is generally a UX security issue.

*enumerateDevices()* does not require permission, so I'd treat that as a separate issue I think.

**Comment by youennf** [comment]
> Having prompts appear from background tabs is generally a UX security issue.

Right, Safari is not resolving getUserMedia promise even in the case where no prompt will be shown to user (say user granted persistent access to this website).

One could think of relaxing this rule in case the device is already capturing.
But that can probably be emulated with clone+applyConstraints so I am not sure this is worth optimizing.

I guess the idea would be to postpone the whole getUserMedia process starting at step 6 until page has focus.

**Comment by jan-ivar** [comment]
> Safari is not resolving getUserMedia promise even in the case where no prompt will be shown to user (say user granted persistent access to this website).

Firefox has the same behavior. Testing with [this fiddle](https://jsfiddle.net/jib1/jyysn48t/) in two tabs, Chrome and (early) Edge do not. @alvestrand @henbos @aboba thoughts?

> relaxing this rule in case the device is already capturing. [ = clone+applyConstraints]

I agree.

> I guess the idea would be to postpone the whole getUserMedia process starting at step 6 until page has focus.

I can submit a PR for discussion. Then we can argue over MAY, SHOULD or MUST, if we like this.
